extension/src/experiments/index.ts

```diff
@@ -579,8 +579,8 @@ export class Experiments extends BaseRepository<TableData> {
   private setupInitialData() {
     const waitForInitialData = this.dispose.track(
       this.onDidChangeExperiments(async () => {
-        await this.pipeline.isReady()
         this.deferred.resolve()
```

Comment:
`getHasData` (how setup checks if experiments is done loading) was returning `undefined` since this function doesn't fire `.resolve()` until the pipeline is ready. I moved `this.diferred.resolve()` to be first which does fix the problem but there's probably a better way to solve this 🤔

Reply:
I'll take a look at this today.

Reply:
I think we will need an event for getHasData changing

Reply:
Adding something like this to the `Setup` constructor is the most obvious for what is needed:

```
    this.getHasData = () => experiments.getHasData()
    this.getExpShowError = () => experiments.getCliError()

    void experiments.isReady().then(() => {
      const updateProjectHasData = () =>
        Promise.all([
          this.sendDataToWebview(),
          setContextValue(ContextKey.PROJECT_HAS_DATA, this.getHasData())
        ])
      void updateProjectHasData()
      const onDidChangeHasData = experiments.columnsChanged.event
      this.dispose.track(onDidChangeHasData(() => updateProjectHasData()))
    })
```
I would pull the new code out into a new method named something like `watchExoerimentHasDataChanges` 

Hope that helps.

Reply:
We also can't initialize experiments before the pipeline is ready because...(had to use GitLens to find out this was added in https://github.com/iterative/vscode-dvc/pull/4264) we need `Pipeline` to have calculated the value for `hasConfig` before we can send anything to the webview.